vllm/config/compilation.py

```diff
@@ -509,7 +516,35 @@ def __post_init__(self) -> None:
                                  "must be 'all', 'none', '+op' or '-op' "
                                  "(where 'op' is the registered op name)")
 
+        # Currently only eager and inductor backend are supported
+        # for piecewise compilation. Update when more backends are supported.
+        if self.level == CompilationLevel.PIECEWISE and self.backend not in [
+                "", "eager", "inductor"
+        ]:
+            raise ValueError(
+                f"Invalid backend for piecewise compilation: {self.backend}")
+
+        if self.backend == "":
+            self.backend = "inductor"
+
+        logger.warning_once(
+            "The 'use_inductor' flag is deprecated and will be\
+                 removed in a future release."
+            "Please use the 'backend' option instead.", )
```

Comment:
I assume this is meant to be under an if statement?

```suggestion
        if self.use_inductor is not None:
            logger.warning_once(
                "The 'use_inductor' flag is deprecated and will be"
                "removed in a future release."
                "Please use the 'backend' option instead.", )
            # TODO check that backend is "" and throw ValueError otherwise
            self.backend = "inductor" if self.use_inductor else "eager"
```

Reply:
After all this, once `backend` is initialized, we should add `'none'` or `'all'` to the `custom_ops` list depending on the backend: `'none'` for level 1-3 if `backend!=eager`, `'all'` otherwise.

Reply:
thanks this was what I was missing. I didn't know what the desired behavior for custom ops was given we are no longer having it by set by the prior properties. Hence why I chose the default of ["all". Now, I will update accordingly.